### ISSUE TITLE
Waiting for CRD resource to become ready on timoni apply

### ISSUE THREAD

**Issue by tired-old-man** [stefanprodan/timoni#391]
Timoni **apply** will just pass through while waiting for CRD resource to become ready.

The problem is not that timoni won't wait, but that CRD operator won't be fast enough to set status conditions.

Details:
The apply command (cmd/timoni/apply.go) is using _ResourceManager_ to wait for kubernetes objects to become ready.
The _ResourceManager_ waiting is defined in _fluxcd_ (ssa/manager_wait.go). It will wait for status "Current".
While there are multiple implementations for reading status in _fluxcd_ (deployment, pod, ...), the _generic_ status reader will be used for CRD. This will check for conditions **Reconciling** and **Ready** (cli-utils/kstatus/status/status.go:Compute). If no conditions are set, the computed status is "Current".
If CRD operator didn't had a chance to set any condition, waiting on **apply** for CRD resource to become ready will just pass through.

How to replicate:
Make a simple timoni module that will deploy single CRD object (assuming operator for CRD will set **Reconciling** or **Ready** condition).
Execute: `timoni apply foo .`
This will pass through. Then repeat the command again: `timoni apply foo .`
The output will show object to be "unchanged", but this time timoni will wait for the CRD resource to become ready.

The _dirty and incorrect_ fix:
Sleep for some time before waiting for kubernetes objects to become ready.
`time.Sleep(time.Duration(500) * time.Millisecond)` after line 326 in `cmd/timoni/apply.go` before _ResourceManager_ Wait call.
Of course this is not correct solution, but will give a chance for operator to set status conditions.

Why not set conditions _before_:
The only other way to set conditions is in _MutatingAdmissionWebhook_.
This is not allowed and it will be ignored by kubernetes.

**Comment by stefanprodan** [comment]
Yes this is known issue with `kstatus`, in Flux CRDs for example we've set `observedGeneration` default to `-1` [here] (https://github.com/fluxcd/kustomize-controller/blob/0007a71e6c36619a226204d92468b4a908bea253/api/v1/kustomization_types.go#L320) and this makes kstatus wait for the controller to reconcile the resources and bump the generation to 0.

**Comment by cgiraldo** [comment]
I think I am hitting this.

In my case I have a timoni module with a "bundle.trust.cert-manager.io" CRD, but its healthy condition is "Synced" instead of "Ready", so timoni returns:  `ERR timeout waiting for: [Bundle/cert-manager/custom-ca status: 'Unknown']`

Do you know any workaround? Thanks

**Comment by stefanprodan** [comment]
@cgiraldo a workaround is to disable waiting for the cert-manager resources, implemented in #462
